src/utils/daos_autotest.c

```diff
@@ -0,0 +1,852 @@
+/**
+ * (C) Copyright 2020 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+
+/* daos_autotest.c - smoke tests against an existing pool to verify installation
+ * invoked by daos(8) utility
+ */
+
+#define D_LOGFAC	DD_FAC(client)
+
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdarg.h>
+
+#include <daos.h>
+#include <daos/common.h>
+
+#include "daos_hdlr.h"
+
+/** Input arguments passed to daos utility */
+struct cmd_args_s *autotest_ap;
+
+/** How many concurrent I/O in flight */
+#define MAX_INFLIGHT 16
+
+#define UUID_STR_SIZE 37	/* 36 + 1 for '\0' */
+char uuid_str[UUID_STR_SIZE];
+
+/** Step timing  */
+clock_t	start;
+clock_t	end;
+
+/** generated container UUID */
+uuid_t		cuuid;
+/** initial object ID */
+uint64_t	oid_hi = 1;
+daos_obj_id_t	oid = { .hi = 1, .lo = 1 }; /** object ID */
+
+/** pool handle */
+daos_handle_t	poh = DAOS_HDL_INVAL;
+/** container handle */
+daos_handle_t	coh = DAOS_HDL_INVAL;
+
+/** force cleanup */
+int force;
+
+static inline void
+new_oid(void)
+{
+	oid.hi = ++oid_hi;
+	oid.lo = 1;
+}
+
+static inline float
+duration(void)
+{
+	return (float) (end - start) * 1E-6;
+}
+
+static inline void
+step_success(const char *comment, ...)
+{
+	va_list	ap;
+	char	timing[8];
+	int	i;
+
+	end = clock();
+	printf("  \033[0;32mOK\033[0m    ");
+	sprintf(timing, "%03.3f", duration());
+	for (i = strlen(timing); i < 7; i++)
+		printf(" ");
+	printf("%s  ", timing);
+	va_start(ap, comment);
+	vprintf(comment, ap);
+	va_end(ap);
+	printf("\n");
+}
+
+static inline void
+step_fail(const char *comment, ...)
+{
+	va_list	ap;
+	char	timing[8];
+	int	i;
+
+	end = clock();
+	printf("  \033[0;31mKO\033[0m    ");
+	sprintf(timing, "%03.3f", duration());
+	for (i = strlen(timing); i < 7; i++)
+		printf(" ");
+	printf("%s  ", timing);
+	va_start(ap, comment);
+	vprintf(comment, ap);
+	va_end(ap);
+	printf("\n");
+
+}
```

Comment:
May be these 2 routines could be combined as a single one as they use the same code, only the banner printf differs !!...

Reply:
done